library/ssl_tls13_client.c

```diff
@@ -200,7 +200,7 @@ static int ssl_tls13_reset_key_share(mbedtls_ssl_context *ssl)
         ssl->handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT;
         return 0;
     } else
-#endif /* MBEDTLS_ECDH_C */
+#endif /* MBEDTLS_PK_CAN_ECDH */
```

Comment:
I think the guard for TLS 1.3 should be `PSA_WANT_ALG_ECDH` actually: unless I'm mistaken, TLS 1.3 use only PSA functions for ECDH, not `mbedtls_ecdh_xxx()` functions. So, if `MBEDTLS_ECDH_C` is enabled but `PSA_WANT_ALG_ECDH` is not (which hardly makes sense but is allowed so far), TLS 1.3 will not be able to do ECDH. As a result, I think we really want to check specifically for `PSA_WANT_ALG_ECDH` here - and in all other places related to TLS 1.3